From: Alexander Graf <agraf@suse.de>
Date: Weg, 20 Mar 2013 16:55:25 +0100
Subject: [PATCH] ARM: FEC: Fix module compilation
Patch-Mainline: No

When comiling FEC as a module, fec.o and fec_ptp.o get
compiled as separate .ko objects. However, fec.o links
to fec_ptp.o and thus requires either EXPORT_SYMBOLs or
to live in the same name space.

Since one module doesn't make sense without the other,
let'd just combine them into a single .ko object.

Signed-off-by: Alexander Graf <agraf@suse.de>

Index: linux-3.9-rc3-master/drivers/net/ethernet/freescale/Makefile
===================================================================
--- linux-3.9-rc3-master.orig/drivers/net/ethernet/freescale/Makefile
+++ linux-3.9-rc3-master/drivers/net/ethernet/freescale/Makefile
@@ -2,7 +2,8 @@
 # Makefile for the Freescale network device drivers.
 #
 
-obj-$(CONFIG_FEC) += fec.o fec_ptp.o
+fecm-objs := fec.o fec_ptp.o
+obj-$(CONFIG_FEC) += fecm.o
 obj-$(CONFIG_FEC_MPC52xx) += fec_mpc52xx.o
 ifeq ($(CONFIG_FEC_MPC52xx_MDIO),y)
 	obj-$(CONFIG_FEC_MPC52xx) += fec_mpc52xx_phy.o
